File: kevoree-tools/org.kevoree.tools.javase.framework/src/main/scala/org/kevoree/framework/message/Message.scala
/**
 * Licensed under the GNU LESSER GENERAL PUBLIC LICENSE, Version 3, 29 June 2007;
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.gnu.org/licenses/lgpl-3.0.txt
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
/**
 * Licensed under the GNU LESSER GENERAL PUBLIC LICENSE, Version 3, 29 June 2007;
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * 	http://www.gnu.org/licenses/lgpl-3.0.txt
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.kevoree.framework.message

/**
 * Licensed under the GNU LESSER GENERAL PUBLIC LICENSE, Version 3, 29 June 2007;
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.gnu.org/licenses/lgpl-3.0.txt
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import java.util.UUID

class Message extends Serializable {

  var destNodeName = "default"

  def getDestNodeName: String = {
    destNodeName
  }

  def setDestNodeName (newDestNodeName: String) {
    destNodeName = newDestNodeName
  }

  var destChannelName = "default"

  def getDestChannelName: String = {
    destChannelName
  }

  def setDestChannelName (newDestChannelName: String) {
    destChannelName = newDestChannelName
  }

  var content: Any = null

  def getContent: Any = {
    content
  }

  def setContent (newContent: Any) {
    content = newContent
  }

  var contentClass: String = null

  def getContentClass: String = {
    contentClass
  }

  def setContentClass (newContentClass: String) {
    contentClass = newContentClass
  }

  var inOut: java.lang.Boolean = false

  def getInOut: java.lang.Boolean = {
    inOut
  }

  def setInOut (newInOut: java.lang.Boolean) {
    inOut = newInOut
  }

  var responseTag = ""

  def getResponseTag: String = {
    responseTag
  }

  def setResponseTag (newResponseTag: String) {
    responseTag = newResponseTag
  }

  var timeout: Long = 3000

  def getTimeout: Long = {
    timeout
  }

  def setTimeout (newTimeout: Long) {
    timeout = newTimeout
  }

  var passedNodes: java.util.List[String] = new java.util.ArrayList[String]()

  def getPassedNodes: java.util.List[String] = {
    passedNodes
  }

  def setPassedNodes (newPassedNodes: java.util.List[String]) {
    passedNodes = newPassedNodes
  }

  var uuid: UUID = UUID.randomUUID()

  def getUuid: UUID = {
    uuid
  }

  def setUuid (newUuid: UUID) {
    uuid = newUuid
  }

  override def clone: Message = {
    val clone = new Message
    clone.setDestNodeName(this.getDestNodeName)
    clone.setDestChannelName(this.getDestChannelName)
    clone.setContent(this.getContent)
    clone.setContentClass(this.getContentClass)
    clone.setInOut(this.getInOut)
    clone.setResponseTag(this.getResponseTag)
    clone.setTimeout(this.getTimeout)
    clone.setPassedNodes(this.getPassedNodes)
    clone.setUuid(this.getUuid)
    clone
  }


}

File: kevoree-tools/org.kevoree.tools.javase.framework/src/main/java/org/kevoree/framework/message/Message.kt
package org.kevoree.framework.message

import java.io.Serializable
import java.util.UUID

class Message: Serializable {

    var _destNodeName = "default"

    fun getDestNodeName(): String {
        return _destNodeName
    }

    fun setDestNodeName(newDestNodeName: String) {
        _destNodeName = newDestNodeName
    }

    var _destChannelName = "default"

    fun getDestChannelName(): String {
        return _destChannelName
    }

    fun setDestChannelName (newDestChannelName: String) {
        _destChannelName = newDestChannelName
    }

    var _content: Any? = null

    fun getContent(): Any? {
        return _content
    }

    fun setContent (newContent: Any) {
        _content = newContent
    }

    var _contentClass: String? = null

    fun getContentClass(): String? {
        return _contentClass
    }

    fun setContentClass (newContentClass: String) {
        _contentClass = newContentClass
    }

    var _inOut: Boolean = false

    fun getInOut(): Boolean {
        return _inOut
    }

    fun setInOut (newInOut: Boolean) {
        _inOut = newInOut
    }

    var _responseTag = ""

    fun getResponseTag(): String {
        return _responseTag
    }

    fun setResponseTag (newResponseTag: String) {
        _responseTag = newResponseTag
    }

    var _timeout: Long = 3000

    fun getTimeout(): Long {
        return _timeout
    }

    fun setTimeout (newTimeout: Long) {
        _timeout = newTimeout
    }

    var _passedNodes: List<String> = java.util.ArrayList<String>()

    fun getPassedNodes(): List<String> {
        return _passedNodes
    }

    fun setPassedNodes (newPassedNodes: List<String>) {
        _passedNodes = newPassedNodes
    }

    var _uuid: UUID = UUID.randomUUID()

    fun getUuid(): UUID {
        return _uuid
    }

    fun setUuid (newUuid: UUID) {
        _uuid = newUuid
    }

    override public fun clone(): Message {
        val clone = Message()
        clone.setDestNodeName(this.getDestNodeName())
        clone.setDestChannelName(this.getDestChannelName())
        clone.setContent(this.getContent())
        clone.setContentClass(this.getContentClass())
        clone.setInOut(this.getInOut())
        clone.setResponseTag(this.getResponseTag())
        clone.setTimeout(this.getTimeout())
        clone.setPassedNodes(this.getPassedNodes())
        clone.setUuid(this.getUuid())
        return clone
    }


}

